packages/stateless/components/inputs/AddressInput.tsx

```diff
@@ -145,18 +145,28 @@ export const AddressInput = <
     return () => document.removeEventListener('keydown', handleKeyPress)
   }, [autofillProfiles, selectAutofillProfile, showProfileAutofill])
 
+  // Only display entity if input is disabled and we're showing the entity. This
+  // is probably showing in a readonly form with submitted data.
+  const onlyDisplayEntity = disabled && showEntity
+
   return (
     <div
       className={clsx(
-        'secondary-text group relative flex items-center gap-3 rounded-md bg-transparent py-3 px-4 font-mono text-sm ring-1 transition-all focus-within:ring-2',
-        error && !showProfileAutofill
-          ? 'ring-border-interactive-error'
-          : 'ring-border-primary focus:ring-border-interactive-focus',
+        'secondary-text group relative flex min-w-0 items-center gap-3 bg-transparent font-mono text-sm transition-all',
+        // If not only displaying entity, add more border.
+        onlyDisplayEntity
+          ? 'p-2'
+          : [
+              'rounded-md py-3 px-4 ring-1 focus-within:ring-2 ',
+              error && !showProfileAutofill
+                ? 'ring-border-interactive-error'
+                : 'ring-border-primary focus:ring-border-interactive-focus',
+            ],
```

Comment:
should we hide the error outline if `onlyDisplayEntity` is true?

Reply:
yep! it's in that array in the else block of the ternary for that reason :)